### PR TITLE
Cleanup Filtering

### PR DESCRIPTION
**Pull request by brownd1978**
Rationalize the DetStepFilters in primary production.  Add a generic bash script for generating primary fcl.  Cleanup the cosmic scripts.  Remove the cosmic 'loose' stream: this needs to become a dedicated S2 job for Offspill and Extracted.

### CODE REVIEW

**Comment by soleti** [comment]
@brownd1978 it looks like the validation failed, do you know the reason?

**Comment by kutschke** [comment]
The job ceSteps is writing an output file named: dts.owner.CeEndpoint.version.sequencer.art

The jobs ceDigi and ceMix fail because they cannot find an input file named: dts.owner.ceSteps.dsconf.seq.art

If I look at previous CI jobs that worked, the output of ceSteps was named: dts.owner.ceSteps.dsconf.seq.art


From: Stefano Roberto Soleti ***@***.***>
Reply-To: Mu2e/Production ***@***.***>
Date: Friday, November 26, 2021 at 9:31 PM
To: Mu2e/Production ***@***.***>
Cc: Rob Kutschke ***@***.***>, Team mention ***@***.***>
Subject: Re: [Mu2e/Production] Cleanup Filtering (PR #141)


@brownd1978<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_brownd1978&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=9s0n95fodLptYW9rwpGMZbAgTLY7Ac7_ttu2fTvquJBbH1auzhKVi_3Vp7SyDWKq&s=sR-WjQBVoLjPO6n0RM1RdH8xcZXqmdaVWiUBWY6n4Pg&e=> it looks like the validation failed, do you know the reason?

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_Mu2e_Production_pull_141-23issuecomment-2D980489004&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=9s0n95fodLptYW9rwpGMZbAgTLY7Ac7_ttu2fTvquJBbH1auzhKVi_3Vp7SyDWKq&s=976CXSm5IjdEmJLoz4d9FSFCVSTpoBlN236rRQliiiI&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ABHY5ZWB4F3X6GU244WSXRLUOA7F7ANCNFSM5IWZGF5Q&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=9s0n95fodLptYW9rwpGMZbAgTLY7Ac7_ttu2fTvquJBbH1auzhKVi_3Vp7SyDWKq&s=ev_aEkQs0iBPEaqyjkeUOxJcnfMS1QdS-FGHtog-VaY&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=9s0n95fodLptYW9rwpGMZbAgTLY7Ac7_ttu2fTvquJBbH1auzhKVi_3Vp7SyDWKq&s=6HFKJJyhypsbPe-bB5Nj3kWOmR3fB015HQYzpSyAvXc&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=gRgGjJ3BkIsb5y6s49QqsA&r=Zi5IH-k2_JiK_Wipiv3pwwSXO9UNaHiLP8sisfPz__k&m=9s0n95fodLptYW9rwpGMZbAgTLY7Ac7_ttu2fTvquJBbH1auzhKVi_3Vp7SyDWKq&s=wfCqxXvB42EcxHRfMM2MxoGTmDbz_NXZaYtEMKTDxd0&e=>.

**Comment by brownd1978** [comment]
@FNALbuild run build test

**Review by soleti — APPROVED** [review]

File: JobConfig/cosmic/prolog.fcl [hunk 1]
```diff
@@ -110,17 +110,6 @@ Cosmic: {
     mcTrajectoryInputs : [ "g4run" ]
     vetoDaughters: []
   }
-  signalFilter : {
```
Comment (oksuzian) [inline JobConfig/cosmic/prolog.fcl:113]: I don't understand where this variable would be defined if we delete it

File: JobConfig/cosmic/S1DSStops.fcl [hunk 2]
```diff
@@ -29,15 +28,27 @@ physics: {
 
   # Here we moved the genCounter at the end in order to get a correct scaling of the
   # cosmic livetime after resampling
-  PrimaryPath :  [generate, @sequence::Common.g4Sequence, stepPointMomentumFilter, cosmicFilter, compressPV, genCounter]
-  EndPath : @local::Primary.EndPath
-
+  S1Path :  [generate, @sequence::Common.g4Sequence, stepPointMomentumFilter, cosmicFilter, compressPV, genCounter]
+  EndPath : [ genCountLogger, S1Output ]
```
Comment (oksuzian) [inline JobConfig/cosmic/S1DSStops.fcl:32]: Should it be S1Output->S1Path

**Review by oksuzian — CHANGES_REQUESTED** [review]

**Review by oksuzian — APPROVED** [review]

**Pull request merged**